pydanticrud/backends/dynamodb.py

```diff
@@ -144,10 +144,15 @@ def __init__(self, cls):
         )
 
     def _serialize_field(self, field_name, value):
+        definition = self.schema["definitions"]
+        schema = self.schema["properties"]
+        for k, v in definition.items():
+            schema[k.lower()] = v
         schema = self.schema["properties"]
         field_type = schema[field_name].get("type", "anyOf")
         try:
-            return SERIALIZE_MAP[field_type](value)
+            if field_name in self.schema['required'] and value is not None:
+                return SERIALIZE_MAP[field_type](value)
```

Comment:
If I'm reading this correctly. Fields that are not required are not serialized at all. If the conditional at 154 fails, then this function returns `None`. I'm pretty sure that's not what we want.

Reply:
yeah ,if field is not in required and value is not None there is no need to serialize that right?

Reply:
I need to write a test case for this scenario

Reply:
test cases added